eclair-core/src/main/scala/fr/acinq/eclair/io/Peer.scala

```diff
@@ -630,17 +630,23 @@ object Peer {
   // @formatter:on
 
   def makeChannelParams(nodeParams: NodeParams, defaultFinalScriptPubKey: ByteVector, isFunder: Boolean, fundingAmount: Satoshi): LocalParams = {
-    val entropy = new Array[Byte](16)
+    val entropy = new Array[Byte](8)
     secureRandom.nextBytes(entropy)
     val bis = new ByteArrayInputStream(entropy)
-    val channelKeyPath = DeterministicWallet.KeyPath(Seq(Protocol.uint32(bis, ByteOrder.BIG_ENDIAN), Protocol.uint32(bis, ByteOrder.BIG_ENDIAN), Protocol.uint32(bis, ByteOrder.BIG_ENDIAN), Protocol.uint32(bis, ByteOrder.BIG_ENDIAN)))
-    makeChannelParams(nodeParams, defaultFinalScriptPubKey, isFunder, fundingAmount, channelKeyPath)
+    // we make sure that funder and fundee key path end differently
+    val last = isFunder match {
+      case false => DeterministicWallet.hardened(0)
+      case true => DeterministicWallet.hardened(1)
+    }
+    val fundingKeyPath = DeterministicWallet.KeyPath(Seq(Protocol.uint32(bis, ByteOrder.BIG_ENDIAN), Protocol.uint32(bis, ByteOrder.BIG_ENDIAN), last))
```

Comment:
Why do we reduce the depth of the funding key path compared to before?

Reply:
What we want to avoid here is reusing keys (either funding keys, or channel keys that are derived from them) and I think 8 bytes (i.e. 2^64 keys) is enough.